### ISSUE TITLE
both collection(reference, path, pathSegments) points to the same place

### ISSUE THREAD

**Issue by marcusx2** [firebase/firebase-js-sdk#7667]
### Operating System

macOS

### Browser Version

Safari 16.5.2

### Firebase SDK Version

10.4.0

### Firebase SDK Product:

Firestore

### Describe your project's tooling

index.html with source tag.

### Describe the problem

The documentation has 2 links for the collection method that points to the same collection [method overload](https://firebase.google.com/docs/reference/js/firestore_.md#collection). 

<img width="1136" alt="image" src="https://github.com/firebase/firebase-js-sdk/assets/5656275/f197dfcf-065b-40fb-9bc3-51e45cf4e9d2">

By inspecting the source, there seems to be 3 variants though

```
export declare function collection(
  firestore: FirebaseFirestore_2,
  path: string,
  ...pathSegments: string[]
): CollectionReference<DocumentData>;

export declare function collection(
  reference: CollectionReference<unknown>,
  path: string,
  ...pathSegments: string[]
): CollectionReference<DocumentData>;

export declare function collection(
  reference: DocumentReference,
  path: string,
  ...pathSegments: string[]
): CollectionReference<DocumentData>;
```

I don't know if this is just a documentation issue. Is it intended that the end user only has access to one of the variants? Also, is it possible to retrieve any document collection only by its `id`, regardless of the variant used?

### Steps and code to reproduce issue

Already described.

**Comment by MarkDuckworth** [comment]
It looks like the documentation is missing the overloads, we will get this fixed, thanks. Can you also send a link to the page with your first screenshot, I don't see that in my browser.

> Also, is it possible to retrieve any document collection only by its id, regardless of the variant used?

I'm not sure if I understand the question. Are you asking for something like the following.
```
const db = initializeFirestore(firebaseApp, options);
const coll = collection(db, "/path/to/collection/by/id");
```

**Comment by marcusx2** [comment]
I can't link it, this is the closest you [can get to](https://firebase.google.com/docs/reference/js/firestore_.md#@firebase/firestore).  

> I'm not sure if I understand the question. Are you asking for something like the following.

The collection has [an id](https://firebase.google.com/docs/reference/js/firestore_.collectionreference.md#properties). If I have this id, can I always retrieve any collection with `collection(db, id);`, or something? I guess a better question is if the id is always equal to the path. Because the collection method takes a path as an argument, but from my tests it seems to be the same thing as the id, so I don't know.

**Comment by marcusx2** [comment]
I'd also like to ask for examples using the DocumentReference and CollectionReference overloads. They don't seem to work at all, or I'm failing to understand how to use them. There are no examples. I always get this error `TypeError: n.indexOf is not a function. (In 'n.indexOf("//")', 'n.indexOf' is undefined)`. I can only successfully use the overload that is [already documented ](https://firebase.google.com/docs/reference/js/firestore_.md#collection).

**Comment by marcusx2** [comment]
I'd like to point out that there is the same issue for the doc method

<img width="889" alt="image" src="https://github.com/firebase/firebase-js-sdk/assets/5656275/ecb04424-1e80-41e4-92ef-b77141d2bd75">,

the onSnapshot method

<img width="889" alt="image" src="https://github.com/firebase/firebase-js-sdk/assets/5656275/b1fdfa1d-019f-4077-9cbe-9c56230f4489">


the setDoc method has 2 reference overloads, but both links point to the same overload

<img width="889" alt="image" src="https://github.com/firebase/firebase-js-sdk/assets/5656275/c765561c-af28-4083-b825-da81fb3da1ed">


Same issue with updateDoc method...

<img width="889" alt="image" src="https://github.com/firebase/firebase-js-sdk/assets/5656275/fbfc3774-a956-4a1b-a5c5-d21a862305de">



Basically the docs for the methods with reference parameter need to be revised, many of them are wrong...
